data/core/init.lua

```diff
@@ -1383,16 +1384,16 @@ function core.step()
   -- update window title
   local current_title = get_title_filename(core.active_view)
   if current_title ~= nil and current_title ~= core.window_title then
-    system.set_window_title(core.compose_window_title(current_title))
+    system.set_window_title(core.window, core.compose_window_title(current_title))
     core.window_title = current_title
   end
 
   -- draw
-  renderer.begin_frame()
+  renderer.begin_frame(core.window)
```

Comment:
Making a review comment to consider what Guldo said so replies can be focused.

I'm okay with simply setting a global active window when passed to `renderer.begin_frame` (with end_frame unsetting the global and doing sanity checks)

What about the other functions though, which aren't directly rendering related?
(such as `system.set_window_title`)?

Reply:
Okay, all the rendering code no longer takes a window as a parameter and instead works with the active window for the current frame.

I'm probably going to move over the `system` window functions into `renwindow` and remove them completely since they are a lot more flexible and not suited for this.